### DISCUSSION TITLE
Allow "MAIN" shell to be dynamic

### DISCUSSION THREAD

**Discussion by wonhyukchoi** [GitSquared/edex-ui#753] [Ideas]
**Perceived idea awesomeness**
- [ ] Wow, you don't have this yet???
- [X] Would be super cool.
- [ ] I'm not sure a lot of people would use it, but I'd love to have it.
- [ ] Kind of a gadget, honestly, but why not?

**Perceived idea difficulty**
- [ ] Quit your job first.
- [X] Would need some work, but nothing impossible.
- [ ] Shouldn't be too hard to make.
- [ ] Like, *I* could do it.

---

My request is twofold:

1. When we exit out of a shell, we default back to the MAIN shell, instead of the shell "just right of it" like it does in a standard terminal. Allowing shell exits to redirected to the "shell right next to it" might be convenient.

2. When we exit out of the MAIN shell, this causes the program to exit even if we have other shells alive. It'd be nice to dynamically re-assign this "MAIN" shell, or perhaps, eliminate the need for a MAIN shell ;)

Thank you!

**Comment by danbulant** [comment]
Probably the easiest to do would be to check for other active terminals and just move them one index to the left (or until another shell takes the place of main). Like if you close the first browser tab.
